### PR TITLE
Add support for PCIe cellular modems

### PR DESCRIPTION
**Pull request by milan-zededa**
With modern high-speed 4G and 5G modems, it is more and more common to connect the modem over the PCIe bus instead of the USB bus. PCIe offers higher speed, lower latency and lower power consumption than USB equivalent, making it perfectly suitable for 5G high speed requirements (up to 20Gbps).

To support PCIe modems, we first had to enable all kernel modules related to the [Qualcomm Modem-Host Interface (MHI)](https://docs.kernel.org/mhi/mhi.html), which was done in the eve-kernel repo, e.g.: https://github.com/lf-edge/eve-kernel/pull/91, and bring these kernel modules into eve: https://github.com/lf-edge/eve/pull/3846

Next, some improvements were necessary in the mmagent to support referencing modems by the PCI address (instead of USB address). This, together with a documentation update, is done in this commit.

### CODE REVIEW

File: pkg/wwan/mm-init.sh [hunk 1]
```diff
@@ -23,7 +23,8 @@ enable_fcc_unlock() {
 }
 
 echo "Loading kernel modules used by ModemManager"
-modprobe -a qcserial usb_wwan qmi_wwan cdc_wdm cdc_mbim cdc_acm
+modprobe -a qcserial usb_wwan qmi_wwan cdc_wdm cdc_mbim cdc_acm \
+            wwan mhi mhi_pci_generic mhi_wwan_ctrl mhi_wwan_mbim
```
Comment (eriknordmark) [inline pkg/wwan/mm-init.sh:27]: Does the addition of these modules cause any significant size increase in the running kernel memory usage?

File: pkg/wwan/mmagent/mmdbus/client.go [hunk 2]
```diff
@@ -749,8 +749,14 @@ func (c *Client) getModemPhysAddrs(modemObj dbus.BusObject) (
...
 	devPath, err := filepath.EvalSymlinks(devPathSymlink)
 	if err != nil {
-		return addrs, proto, fmt.Errorf("failed to eval symlink %s: %w",
-			devPathSymlink, err)
+		// Maybe this modem uses PCIe interface as opposed to USB.
+		// This is more and more common for modern high-speed 4G and 5G modems.
+		devPathSymlink = filepath.Join("/sys/class/wwan", primaryPort, "device")
+		devPath, err = filepath.EvalSymlinks(devPathSymlink)
+		if err != nil {
+			return addrs, proto, fmt.Errorf("failed to eval symlink %s: %w",
```
Comment (eriknordmark) [inline pkg/wwan/mmagent/mmdbus/client.go:757]: This just prints the PCIe path. Would it make sense to include both paths which were tried and failed in the error string?

File: pkg/wwan/mmagent/mmdbus/client.go [hunk 3]
```diff
@@ -825,6 +816,28 @@ func (c *Client) getModemPhysAddrs(modemObj dbus.BusObject) (
 	return addrs, proto, nil
 }
 
+// Given device path from /sys, such as: /sys/devices/pci0000:00/0000:00:1c.0/0000:2c:00.0/mhi0/wwan/wwan0
+// it returns the address of the device on the given bus (e.g. "0000:2c:00.0" for "pci").
```
Comment (eriknordmark) [inline pkg/wwan/mmagent/mmdbus/client.go:820]: Might make sense to add a comment that if the device is not on the particular bus, it returns an empty string (and not an error).

**Review by eriknordmark — APPROVED** [review]
Some nits. Kick off tests.

File: pkg/wwan/mm-init.sh (hunk 1 above)
Comment (milan-zededa) [reply pkg/wwan/mm-init.sh:27]: I checked `/proc/modules` and `/proc/meminfo` between 12.0.0 and this PR, but do not see any significant kernel memory usage increase.

File: pkg/wwan/mmagent/mmdbus/client.go (hunk 2 above)
Comment (milan-zededa) [reply pkg/wwan/mmagent/mmdbus/client.go]: Makes sense, done.

File: pkg/wwan/mmagent/mmdbus/client.go (hunk 3 above)
Comment (milan-zededa) [reply pkg/wwan/mmagent/mmdbus/client.go:821]: Done

File: pkg/wwan/mm-init.sh (hunk 1 above)
Comment (eriknordmark) [reply pkg/wwan/mm-init.sh:27]: Thanks

**Pull request merged**